include/boost/gil/image_processing/threshold.hpp

```diff
@@ -0,0 +1,85 @@
+//
+// Copyright 2019 Miral Shah <miralshah2211@gmail.com>
+//
+// Use, modification and distribution are subject to the Boost Software License,
+// Version 1.0. (See accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt)
+//
+#ifndef BOOST_GIL_THRESHOLD_HPP
+#define BOOST_GIL_THRESHOLD_HPP
+
+#include <functional>
+#include <boost/gil/image.hpp>
+
+namespace boost { namespace gil {
+
+
+namespace detail {
+
+template <typename source_channel_t, typename result_channel_t,
```

Comment:
## Formatting

```
template <typename source_channel_t, typename result_channel_t,
    typename SrcView, typename DstView>
```
should rather read
```
template <typename source_channel_t, typename result_channel_t, typename SrcView, typename DstView>
```

but only if single line is not longer than 90 characters, or
```
template
<
    typename source_channel_t,
    typename result_channel_t,
    typename SrcView,
    typename DstView
>
```

## Naming

Template parameters should be `CamelCase`, not `source_channel_t`